src/OpenCvSharp.Extensions/BitmapConverter.cs

```diff
@@ -156,8 +155,8 @@ public static unsafe void ToMat(this Bitmap src, Mat dst)
                             throw new ArgumentException("Invalid nChannels");
                         if (dst.Depth() != MatType.CV_8U && dst.Depth() != MatType.CV_8S)
                             throw new ArgumentException("Invalid depth of dst Mat");
-                        if (src.Palette.Flags == 4) // https://docs.microsoft.com/ja-jp/dotnet/api/system.drawing.imaging.colorpalette.flags?view=netframework-4.8
-                            throw new NotImplementedException("Not supported halftone Palette");
+                        //if (src.Palette.Flags == 4) // https://docs.microsoft.com/ja-jp/dotnet/api/system.drawing.imaging.colorpalette.flags?view=netframework-4.8
+                        //    throw new NotImplementedException("Not supported halftone Palette");
```

Comment:
I assume that this check is guarding against trying to use any bitmap except of grayscale (as only red component is used from palette), right? Maybe there is another way to make sure that the bitmap is grayscale to still be able to throw on unexpected input? For example, check that all colors in palette have same R, G and B value. But I'm not sure how it will impact conversion performance.

Reply:
Thank you for your comment. This code block is for Bitmap.PixelFormat==8bppIndexed. I understand 8bppIndexed to mean grayscale. So if the values of each of the palette's R, G, and B are different, I'm not sure how to handle that.
```c#
// pseudo-code
byte[] bitmapValues = ... ;
var palette = bitmap.Palette;

byte[] matValues = bitmapValues.Select(b => {
    byte newR = palette.Entries[b].R;
    byte newG = palette.Entries[b].G;
    byte newB = palette.Entries[b].B;
    return ???; // What should I merge RGB into one byte value?
}).ToArray();
```

In the EmguCV implementation, it looks like that the method returns a 8UC3 (Color) Mat, even though the input Bitmap is 8bppIndexed. That makes me wonder a little bit, but maybe I should go with that.

Reply:
I did some more research and here's what I got.
`8bppIndexed` is in fact not defined as grayscale format, it just defines that there are 256 colors in palette, which consists of any colors. [Here ](https://gist.github.com/kanstantsin/b5518b2505f5f3dd3d08cfc85826ef52)is the code which generates 64x64 green and red bitmap with `8bppIndexed` format. One more interesting thing is that `Palette.Flags` contains value `4` regardless of palette contents which is created automatically (you can't create palette manually or change its flags). At least I didn't find a way to create `8bppIndexed` image with other values in flags. Perhaps it is possible when bitmap is loaded from file.
What I think about it is probably we can check destination `Mat` type and act depending on number of channels in it. As a simple solution maybe we can stay with current version of merge request (although I prefer removing code instead of commenting it out in production code), but keep in mind that current version of code does not cover colored `8bppIndexed` images.

Reply:
Thank you for the useful information. I made a branch of the process by type of dst.